### ISSUE TITLE
Extension needs to be updated for GNOME 45

### ISSUE THREAD

**Issue by fmarier** [rcaelers/workrave#508]
This was originally reported to me on <https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1052115> (also on [Ubuntu](https://bugs.launchpad.net/ubuntu/+source/gnome-shell-pomodoro/+bug/2031918)). The GNOME extension will need to be updated to be compatible with GNOME 45. Details are at <https://gjs.guide/extensions/upgrading/gnome-shell-45.html>.

It unfortunately sounds like it's not possible to be compatible with both GNOME 44 and GNOME 45 due to the change in how external libraries are imported. This is not a problem for Debian and its derivatives because we only support the version of GNOME that's in Debian/Ubuntu/etc., but it may be a problem for the upstream workrave project itself.

**Comment by rcaelers** [comment]
Thanks for the heads-up!

I migrated the gnome-shell extension to GNOME 45 on the main (1.11) branch. It seems to work, but I will need to do some more testing. After testing, I will sync the change to 1.10 and create a new release.

Note that I probably can't detect the GNOME version at build time. In my CI pipeline does not have GNOME installed (only libgnome-panel-dev), so I opted for an option (WITH_GNOME45 in CMake). Do you know if GNOME45 can be auto-detected build time in the Debian build?

**Comment by fmarier** [comment]
> Do you know if GNOME45 can be auto-detected build time in the Debian build?

Good. I don't know enough about GNOME to know the answer here. It would certainly be nice to be able to detect it and build for the right version automatically.

**Comment by rcaelers** [comment]
@jbicha Thanks!

For now, I added a '--enable-gnome45' configure option to Workrave, without any auto-detect. I couldn't find any cross-distro way to detect GNOME 45 without installing gnome-shell itself (which Workrave does not need build-time and is quite large).

I solved the GNOME 45 detect in a Debian/Ubuntu specific way. In the 'debian/rules' file for my PPA Workrave package I have:

```
GNOME_SHELL_VERSION := $(shell apt-cache policy gnome-shell | grep Candidate: | awk '{print $$2}' | cut -d. -f1)
ifeq ($(shell [ $(GNOME_SHELL_VERSION) -ge 45 ]; echo $$?),0)
    CONFIGURE_FLAGS += --enable-gnome45
endif

override_dh_auto_configure:
	dh_auto_configure -- \
                .... \
		$(CONFIGURE_FLAGS)
```

Not sure if this is the best way. If the output format of 'apt-cache policy' changes, this will break.
